R/pkg/R/utils.R

```diff
@@ -698,6 +698,21 @@ isSparkRShell <- function() {
   grepl(".*shell\\.R$", Sys.getenv("R_PROFILE_USER"), perl = TRUE)
 }
 
+captureJVMException <- function(e) {
```

Comment:
since there are 3 cases here and their handling seem nontrivial, could you add a test for each of the 3 exception main types?